### ISSUE TITLE
clipping: difference of a sector and a circle.

### ISSUE THREAD

**Issue by aharjula** [jsxgraph/jsxgraph#346]
Tried to construct a sector and clip out a bit from the centre, and the output was somewhat angular, could the curve representing the sector have some odd behaviour related to its resolution when generated for the operation? I know that clipping is a new thing but here is a test case for future development.

https://jsfiddle.net/ghwpqtc5/7/
![Screenshot_2021-03-10_14-37-33](https://user-images.githubusercontent.com/980957/110630563-321d3680-81ae-11eb-8c5d-94bcce1a202f.png)

For me, the interest is for doing this with the same centre point to generate a portion in a doughnut chart and I would not be that surprised if others will have the same idea.

I also tried to build it by first building a ring by cutting a smaller circle from a larger and intersecting that with the sector (also tried with a polygon with same instability), but that seems to have some numerical instability after one moves it around and especially if the angles (points defining the sector) come from the positions of the other moving elements in the graph.

**Comment by alfredwassermann** [comment]
The problem here is that arcs (and sectors) are rendered as Bezier curves. The coordinate points which make the plot angular are the control points of the Bezier curve, since clipping does not work yet with curves of degree three.
I will try to fix that.

**Comment by aharjula** [comment]
Makes sense. Here is another example related to the same objective. It has an animation that shows the move required to get the intersection area to behave badly. If one moves the centre point around one can trigger much more obvious errors. Of course, I am not sure if this is due to the order of updates of the clipped objects or something else but again something to test against.

https://jsfiddle.net/g3jwsaeb/1/
![Screenshot_2021-03-11_14-11-14](https://user-images.githubusercontent.com/980957/110785488-b1287280-8273-11eb-82e4-49035753a984.png)


Could be that I should just draw an arc with a stroke-width that matches what I want instead of trying to build an area to fill.
